app/storage/src/main/java/com/fsck/k9/storage/StoreSchemaDefinition.java

```diff
@@ -229,6 +242,7 @@ private static void dbCreateDatabaseFromScratch(SQLiteDatabase db) {
                 "BEGIN " +
                 "DELETE FROM message_parts WHERE root = OLD.message_part_id; " +
                 "DELETE FROM messages_fulltext WHERE docid = OLD.id; " +
+                "DELETE FROM outbox_state WHERE message_id = OLD.id; " +
```

Comment:
is there a reason we do this with triggers rather than foreign keys? seems a bit odd. (have I asked this before?)

Reply:
Hmm... it seems possible to use foreign keys and [`CASCADE DELETE`](https://stackoverflow.com/q/5890250) in sqlite when using `foreign_keys` PRAGMA... I didn't know that.

Reply:
Of course. We use that in OpenKeychain [all the time](https://github.com/open-keychain/open-keychain/blob/master/OpenKeychain/src/main/sqldelight/org/sufficientlysecure/keychain/UserPackets.sq#L15).

Reply:
Yes, I see you have [foreign keys enabled](https://github.com/open-keychain/open-keychain/blob/master/OpenKeychain/src/main/java/org/sufficientlysecure/keychain/KeychainDatabase.java#L88) but I don't see it in K-9 (it isn't used?).

One way or another I think FKs are better approach and the work to enable them would be a good investment anyway :)

Reply:
The codebase has never used foreign keys. I believe it's because early Android versions shipped with a Sqlite version that didn't support foreign keys.